engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/AtomInteropTest.java

```diff
@@ -0,0 +1,146 @@
+      assertThat("Has two fields", memberNames.size(), is(2));
+      assertThat(
+          "Member names are not qualified", memberNames, containsInAnyOrder("field_1", "field_2"));
+      for (var memberName : memberNames) {
+        var member = myTypeAtom.getMember(memberName);
+        assertThat("Member " + memberName + " should be readable", member, is(notNullValue()));
+        assertThat("All fields are numbers", member.isNumber(), is(true));
+      }
+    }
+  }
+
+  @Test
+  public void atomIsNotMetaObject() {
+    try (var ctx = ContextUtils.createDefaultContext()) {
+      var myTypeAtom =
+          ContextUtils.evalModule(
+              ctx,
+              """
+          type My_Type
+              Cons field_1 field_2
+
+          main =
+              My_Type.Cons 1 2
+          """);
+      assertThat(myTypeAtom.isMetaObject(), is(false));
+      assertThat(myTypeAtom.getMetaObject().getMetaSimpleName(), is("My_Type"));
+    }
+  }
+
+  @Test
+  public void typeHasAnyAsSuperType() {
+    try (var ctx = ContextUtils.createDefaultContext()) {
+      var myTypeAtom =
+          ContextUtils.evalModule(
+              ctx,
+              """
+          type My_Type
+              Cons
+
+          main = My_Type.Cons
+          """);
+      var myType = myTypeAtom.getMetaObject();
+      assertThat(myType.hasMetaParents(), is(true));
+      var metaParents = myType.getMetaParents();
+      assertThat(metaParents.hasArrayElements(), is(true));
+      assertThat("Has just one meta parent - Any", metaParents.getArraySize(), is(1L));
+      var anyType = metaParents.getArrayElement(0);
+      assertThat(anyType.getMetaSimpleName(), is("Any"));
+    }
+  }
+
+  @Test
+  public void atomMembersAreConstructorFields_ManyConstructors() {
+    try (var ctx = ContextUtils.createDefaultContext()) {
+      var myTypeAtom =
+          ContextUtils.evalModule(
+              ctx,
+              """
+          type My_Type
+              Cons_1 f1 f2 f3 f4 f5 f6
+              Cons_2 g1 g2 g3
+              Cons_3 h1 h2 h3 h4 h5 h6 h7 h8 h9
+
+          main = My_Type.Cons_2 "g1" "g2" "g3"
+          """);
+      assertThat(
+          "Member names correspond to constructor field names for a single constructor",
+          myTypeAtom.getMemberKeys(),
+          containsInAnyOrder("g1", "g2", "g3"));
+    }
+  }
+
+  @Test
+  public void typeMembersAreConstructors() {
+    try (var ctx = ContextUtils.createDefaultContext()) {
+      var myType =
+          ContextUtils.evalModule(
+              ctx,
+              """
+          type My_Type
+              Cons_1
+              Cons_2
+
+          main = My_Type
+          """);
+      assertThat("type has constructors as members", myType.hasMembers(), is(true));
+      assertThat(myType.getMemberKeys(), containsInAnyOrder("Cons_1", "Cons_2"));
+      assertThat(
+          "Constructor (type member) is an executable function",
+          myType.getMember("Cons_1").canExecute(),
```

Comment:
This test fails. `Cons_1` is not executable, but should be, as it is a function that can be invoked. This is not related to this PR, but should be addressed in the future.

Reply:
Fixing https://github.com/enso-org/enso/pull/10661/files#r1690196270 will also fix this test. Should be straightforward.